subsys/net/l2/ieee802154/ieee802154_utils.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023 Florian Grandel, Zephyr Project
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief IEEE 802.15.4 internal MAC and PHY Utils Implementation
+ *
+ * All references to the standard in this file cite IEEE 802.15.4-2020.
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(net_ieee802154, CONFIG_NET_L2_IEEE802154_LOG_LEVEL);
+
+#include "ieee802154_utils.h"
+
+/**
+ * PHY utilities
+ */
+
+bool ieee802154_radio_verify_channel(struct net_if *iface, uint16_t channel)
+{
+	struct ieee802154_attr_value value;
+
+	if (channel == IEEE802154_NO_CHANNEL) {
+		return false;
+	}
+
+	if (ieee802154_radio_attr_get(iface, IEEE802154_ATTR_PHY_SUPPORTED_CHANNEL_RANGES,
+				      &value)) {
+		return false;
+	}
+
+	for (int channel_range_index = 0;
+	     channel_range_index < value.phy_supported_channels->num_ranges;
+	     channel_range_index++) {
+		const struct ieee802154_phy_channel_range *const channel_range =
+			&value.phy_supported_channels->ranges[channel_range_index];
+
+		if (channel >= channel_range->from_channel &&
+		    channel <= channel_range->to_channel) {
+			return true;
+		}
+	}
```

Comment:
This can be just `	for (int idx = 0; idx < value.phy_supported_channels->num_ranges; idx++) {`
and before you complain this is just my personal attitude (which it certainly is, and I have learned from contributing a few thousand lines of code to 15.4 drivers from the maintainers), please [RTFCS](https://www.kernel.org/doc/html/latest/process/coding-style.html#naming).

Reply:
If you were to read our coding style, you would find that "In general, follow the [Linux kernel coding style](https://kernel.org/doc/html/latest/process/coding-style.html), with the following exceptions:..." https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style

Further, I wrote "This can be just...", not "This must be...". IMHO, spreading information unnecessarily over a large number of characters is not good practice, either in coding or in writing RFCs or responses.

Reply:
Thanks for the pointer - very useful.